homeassistant/components/hvv_departures/binary_sensor.py

```diff
@@ -0,0 +1,184 @@
+            for elevator in partial_station.get("elevators", []):
+
+                state = elevator.get("state") != "READY"
+                available = elevator.get("state") != "UNKNOWN"
+                label = elevator.get("label")
+                description = elevator.get("description")
+
+                name = f"Elevator {label} at {station_name} ({description})"
+                lines = elevator.get("lines")
+
+                idx = f"{station_name}-{label}-{lines}"
+
+                elevators.update(
+                    {
+                        idx: {
+                            "state": state,
+                            "name": name,
+                            "available": available,
+                            "attributes": {
+                                "cabin_width": elevator.get("cabinWidth"),
+                                "cabin_length": elevator.get("cabinLength"),
+                                "door_width": elevator.get("doorWidth"),
+                                "elevator_type": elevator.get("elevatorType"),
+                                "button_type": elevator.get("buttonType"),
+                                "cause": elevator.get("cause"),
+                                "lines": lines,
+                                ATTR_ATTRIBUTION: ATTRIBUTION,
+                            },
+                        }
+                    }
+                )
+        return elevators
+
+    async def async_update_data():
+        """Fetch data from API endpoint.
+
+        This is the place to pre-process the data to lookup tables
+        so entities can quickly look up their data.
+        """
+
+        payload = {"station": station}
+
+        try:
+            async with async_timeout.timeout(10):
+                return get_elevator_entities_from_station_information(
+                    station_name, await hub.gti.stationInformation(payload)
+                )
+        except InvalidAuth as err:
+            raise UpdateFailed(f"Authentication failed: {err}") from err
+        except ClientConnectorError as err:
+            raise UpdateFailed(f"Network not available: {err}") from err
+        except Exception as err:  # pylint: disable=broad-except
+            raise UpdateFailed(f"Error occurred while fetching data: {err}") from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="hvv_departures.binary_sensor",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(hours=1),
+    )
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+    async_add_entities(
+        HvvDepartureBinarySensor(coordinator, idx, entry)
+        for (idx, ent) in coordinator.data.items()
+    )
+
+
+class HvvDepartureBinarySensor(BinarySensorEntity):
```

Comment:
We can inherit CoordinatorEntity here.
https://github.com/home-assistant/core/blob/62c4e072f5e21ccd057e343c700d2f81a0a50b2e/homeassistant/helpers/update_coordinator.py#L195

Reply:
I had to inherit from both `CoordinatorEntity` and `BinarySensorEntity`.